File: app/src/main/java/com/example/kotlin/mvc/kotlin/base/LazyLoader.java
package com.example.kotlin.mvc.kotlin.base;

public abstract class LazyLoader {

    private boolean isLazyInit;

    public LazyLoader() {

    }

    public void init() {
        if (isLazyInit)
            return;
        lazyInitView();
        isLazyInit = true;
    }

    protected abstract void lazyInitView();

    public boolean isLazyInit() {
        return isLazyInit;
    }

    public void destroy() {

    }

}

File: app/src/main/java/com/example/kotlin/mvc/kotlin/base/LazyLoader.kt
package com.example.kotlin.mvc.kotlin.base

abstract class LazyLoader {
    var isLazyInit = false
        private set

    fun init() {
        if (isLazyInit) return
        lazyInitView()
        isLazyInit = true
    }

    protected abstract fun lazyInitView()
    fun destroy() {}
}